benchmark/benchmark_long.py

```diff
@@ -0,0 +1,166 @@
+                output_len,
+                REQUEST_LATENCY,
+            )
+            self.left -= 1
+            # pring longer space to overwrite the previous when left decrease
+            print("\rdone_request, left %d    " % (self.left), end="")
+        # The last one
+        print("")
+
+
+def main(args: argparse.Namespace):
+    if args.concurrency > args.num_prompts:
+        print("Fix concurrency with num_prompts %d" % (args.num_prompts))
+        args.concurrency = args.num_prompts
+    print(args)
+
+    api_url = f"http://{args.host}:{args.port}/v1/chat/completions"
+    model_uid = args.model_uid
+
+    logger.info("Preparing for benchmark.")
+    tokenizer = get_tokenizer(args.tokenizer, trust_remote_code=args.trust_remote_code)
+    input_requests = generate_sorting_prompts(
+        args.concurrency, args.context_length, args.context_length / 2 - 20, tokenizer
+    )
+
+    logger.info("Benchmark starts.")
+    benchmark_start_time = time.time()
+
+    benchmark = BenchmarkRunner(
+        api_url,
+        model_uid,
+        input_requests,
+        concurrency=args.concurrency,
+    )
+    asyncio.run(benchmark.run())
+    benchmark_end_time = time.time()
+    benchmark_time = benchmark_end_time - benchmark_start_time
+    print(f"Total time: {benchmark_time:.2f} s")
+    print(f"Throughput: {args.num_prompts / benchmark_time:.2f} requests/s")
+
+    # Compute the latency statistics.
+    avg_latency = np.mean([latency for _, _, latency in REQUEST_LATENCY])
+    print(f"Average latency: {avg_latency:.2f} s")
+    avg_per_token_latency = np.mean(
+        [
+            latency / (prompt_len + output_len)
+            for prompt_len, output_len, latency in REQUEST_LATENCY
+        ]
+    )
+    print(f"Average latency per token: {avg_per_token_latency:.2f} s")
+    avg_per_output_token_latency = np.mean(
+        [latency / output_len for _, output_len, latency in REQUEST_LATENCY]
+    )
+    print("Average latency per output token: " f"{avg_per_output_token_latency:.2f} s")
+    average_io_tokens = np.average(
+        [(prompt_len + output_len) for prompt_len, output_len, _ in REQUEST_LATENCY]
+    )
+    print(f"Average io length:" f"{average_io_tokens}")
+    throughput = (
+        sum([output_len for _, output_len, _ in REQUEST_LATENCY]) / benchmark_time
+    )
+    print(f"Throughput: {throughput} tokens/s")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Benchmark the online serving throughput."
+    )
+    parser.add_argument("--host", type=str, default="localhost")
+    parser.add_argument("--port", type=int, default=9997)
+    parser.add_argument(
+        "--tokenizer", type=str, required=True, help="Name or path of the tokenizer."
+    )
+    parser.add_argument(
+        "--context-length", type=int, default=32768, help="model context_length."
+    )
+    parser.add_argument(
+        "--num-prompts", type=int, default=1000, help="Number of prompts to process."
```

Comment:
Ditto.

Reply:
fixed

Reply:
This one does not changed to 100?

Reply:
OK, you can set it to a smaller value.

Reply:
Fixed, and I've modify comment in this file